src/main/java/teammates/logic/core/FeedbackSessionsLogic.java

```diff
@@ -237,15 +238,33 @@ public boolean isFeedbackSessionHasQuestionForStudents(
      *
      * <p> If there is no question for students, the feedback session is completed</p>
      */
-    public boolean isFeedbackSessionCompletedByStudent(FeedbackSessionAttributes fsa, String userEmail) {
-        if (frLogic.hasGiverRespondedForSession(userEmail, fsa.getFeedbackSessionName(), fsa.getCourseId())) {
-            return true;
-        }
-
+    public boolean isFeedbackSessionCompletedByStudent(FeedbackSessionAttributes fsa, String userEmail, String userTeam) {
         String feedbackSessionName = fsa.getFeedbackSessionName();
         String courseId = fsa.getCourseId();
+
+        List<FeedbackQuestionAttributes> allQuestions =
+                fqLogic.getFeedbackQuestionsForStudents(feedbackSessionName, courseId);
         // if there is no question for students, session is complete
-        return !fqLogic.sessionHasQuestions(feedbackSessionName, courseId);
+        if (allQuestions.isEmpty()) {
+            return true;
+        }
+        boolean isAllTeamQuestions = true;
+
+        for (FeedbackQuestionAttributes question : allQuestions) {
+            if (question.getGiverType() == FeedbackParticipantType.STUDENTS) {
+                isAllTeamQuestions = false;
+                break;
+            }
+        }
+
+        if (isAllTeamQuestions) {
+            // if all team questions, session is complete only if someone from team has responded
+            return frLogic.hasGiverRespondedForSession(userTeam, feedbackSessionName, courseId);
+        } else {
+            // if there are individual questions, session is complete only if
+            // the student has responded to the individual questions
+            return frLogic.hasGiverRespondedForSession(userEmail, feedbackSessionName, courseId);
```

Comment:
The `userEmail` param in `hasGiverRespondedForSession` seems overloaded with `userTeam` now. Do you have a better way of phrasing it?

Reply:
I think a more accurate name for `userEmail` param in `hasGiverRespondedForSession` is `giverIdentifier` since from what I understand, email is used for individual responses but team name is used for team responses. It will be good to probably change the `giverEmail` column in `FeedbackResponse` entity to `giverIdentifier` also.

Reply:
Sounds good to me. Let's see how it looks?

Reply:
> It will be good to probably change the `giverEmail` column in `FeedbackResponse` entity to `giverIdentifier` also.

I think this might not be backward compatible... An alternative way would be to get all students in the team and return `true` if at least one of the students in the team has responded for the session.

@moziliar what do you think.

Reply:
Oh sorry, I missed out the part about changing the entity field... @jianhandev you are right. Let's keep the entity untouched. We should only touch the code logic.

Changing the `userEmail` param to `giverIdentifier` or something similar will do. The problem is minor in that a `userTeam` is given as `userEmail` param.

Reply:
>Changing the userEmail param to giverIdentifier or something similar will do. 

Ok sure